module/modmanager/manager.go

```diff
@@ -155,6 +162,7 @@ func (mgr *Manager) add(ctx context.Context, conf config.Module, conn *grpc.Clie
 	}
 
 	mod.registerResources(mgr, mgr.logger)
+	mgr.modules[conf.Name] = mod
```

Comment:
This really belonged at the end of `add`. We shouldn't register resources and add to the `modules` map until we know `add` was successful; that way, we don't have to awkwardly `delete` from the map in the earlier defer block.